### ISSUE TITLE
Atom whitelist attribute error from mbuild Comound

### ISSUE THREAD

**Issue by richardjgowers** [mosdef-hub/foyer#191]
I'm trying to do some basic examples, currently this fails:

```python
import mbuild as mb
import warnings
import foyer

warnings.filterwarnings("ignore", category=DeprecationWarning) 

a = mb.Particle(name='CH3', pos=[0,0,0])
b = mb.Particle(name='O', pos=[0,0,0.1])
c = mb.Particle(name='CH3', pos=[0,0,0.2])

dme = mb.Compound()
dme.add([a, b, c])

dme.add_bond([a, b])
dme.add_bond([b, c])

ff = foyer.Forcefield()
ff.apply(dme)
```

gives:
```python
---------------------------------------------------------------------------
AttributeError                            Traceback (most recent call last)
<ipython-input-18-5b24986a3d4e> in <module>()
----> 1 ff.apply(dme)

~/miniconda3/envs/mda/lib/python3.6/site-packages/foyer/forcefield.py in apply(self, topology, references_file, use_residue_map, assert_angle_params, assert_dihedral_params, assert_improper_params, *args, **kwargs)
    379             positions[:] = np.nan
    380         box_vectors = topology.getPeriodicBoxVectors()
--> 381         topology = self.run_atomtyping(topology, use_residue_map=use_residue_map)
    382         system = self.createSystem(topology, *args, **kwargs)
    383 

~/miniconda3/envs/mda/lib/python3.6/site-packages/foyer/forcefield.py in run_atomtyping(self, topology, use_residue_map)
    457                     if res.name not in residue_map.keys():
    458                         residue = _topology_from_residue(res)
--> 459                         find_atomtypes(residue, forcefield=self)
    460                         residue_map[res.name] = residue
    461 

~/miniconda3/envs/mda/lib/python3.6/site-packages/foyer/atomtyper.py in find_atomtypes(topology, forcefield, max_iter)
     41 
     42     _iterate_rules(rules, topology, max_iter=max_iter)
---> 43     _resolve_atomtypes(topology)
     44 
     45 

~/miniconda3/envs/mda/lib/python3.6/site-packages/foyer/atomtyper.py in _resolve_atomtypes(topology)
     94     """Determine the final atomtypes from the white- and blacklists. """
     95     for atom in topology.atoms():
---> 96         atomtype = [rule_name for rule_name in atom.whitelist - atom.blacklist]
     97         if len(atomtype) == 1:
     98             atom.id = atomtype[0]

AttributeError: 'Atom' object has no attribute 'whitelist'
```

This is with mbuild v0.8.1 and foyer 0.5.1

The docs for apply say that you can use an mbuild Compound, should this work?

**Comment by mikemhenry** [comment]
I think this error message is from when the force field your applying to a compound is missing the atom types in your compound. I'm not sure if foyer has a default force filed, I think you either need to supply one or tell it to use one of the ones it ships with, like this:

```
ff = foyer.Forcefield("trappe-ua")
ff.apply(dme)
```

If you want to use that force field.

**Comment by uppittu11** [comment]
Fixed in #194
